storage/blockchain/DESIGN.md

```diff
@@ -0,0 +1,600 @@
+# Database
+FIXME: This documentation must be updated and moved to the architecture book.
```

Comment:
The old database design document is in here for now.

Will remove after porting to architecture book.